### PR TITLE
[RDY] i18n: Fix language switching not affecting lighting level names

### PR DESCRIPTION
**Pull request by BrettDong**

### CODE REVIEW

File: src/output.cpp [hunk 1]
```diff
@@ -1756,28 +1756,37 @@ get_hp_bar( const int cur_hp, const int max_hp, const bool is_mon )
            ( ratio >  0.0 )            ? strings[10] : strings[11];
 }
 
-std::pair<std::string, nc_color> const &get_light_level( const float light )
+std::pair<std::string, nc_color> const get_light_level( const float light )
```
Comment (BevapDin) [inline src/output.cpp]: Please remove the first const here, the function doesn't need to return a const value.

File: src/output.cpp [hunk 2]
```diff
@@ -1756,28 +1756,37 @@ get_hp_bar( const int cur_hp, const int max_hp, const bool is_mon )
...
     } else if( light_level > 5 ) {
-        return strings[5];
+        light_level = 5;
     }
-
-    return strings[light_level];
+    switch( light_level ) {
+        case 0:
+            return pair_t{ _( "unknown" ), c_pink };
+            break;
```
Comment (BevapDin) [inline src/output.cpp]: Ugh, what's with all those pointless `break`s here. They are never reached anyway, please remove them.

**Comment by BrettDong** [comment]
Code updated.

File: src/output.cpp [hunk 3]
```diff
@@ -1756,28 +1756,31 @@ get_hp_bar( const int cur_hp, const int max_hp, const bool is_mon )
...
-    };
-
-    const int light_level = ceil( light );
+    int light_level = ceil( light );
     if( light_level < 0 ) {
-        return strings[0];
+        light_level = 0;
     } else if( light_level > 5 ) {
-        return strings[5];
+        light_level = 5;
```
Comment (codemime) [inline src/output.cpp]: Optionally: could be shortened to:
```c++
const int light_level = clamp( ceil( light ), 0, 5 );
```

Comment (BrettDong) [inline src/output.cpp]: `std::clamp` is added in the C++17 standard if I remember correctly.

Comment (codemime) [inline src/output.cpp]: Fortunately, we already have it in `cata_utility.h`.

**Comment by BrettDong** [comment]
```
11:53:49 src/output.cpp:1762:29: error: no matching function for call to 'clamp'
11:53:49     const int light_level = clamp( ceil( light ), 0, 5 );
11:53:49                             ^~~~~
11:53:49 src/cata_utility.h:225:13: note: candidate template ignored: deduced conflicting types for parameter 'T' ('double' vs. 'int')
11:53:49 constexpr T clamp( const T &val, const T &min, const T &max )
11:53:49             ^
11:53:49 1 error generated.
```

I have to do an explicit type conversion?

**Comment by Coolthulhu** [comment]
Is there no less-ugly way to do that? Regressing to switches everywhere just to appease translated string extractor sounds like a backwards way to do this.

**Comment by AMurkin** [comment]
JSONify them?

**Comment by BevapDin** [comment]
Why are switches bad?

**Comment by Coolthulhu** [comment]
> Why are switches bad?

More work later on when converting things to jsons, more work if the switch argument ever needs to become a non-`int` type, more lines meaning less readability.
I had some bad experience with overused switches in DDA code because they allow "code as data" approaches that are hard to clean up. For example, in addiction code, the switches allowed mixing `string_format` with pure string.

Not a huge deal, but if there was a nicer solution, I'd much prefer that to regression to code/data mix.

**Comment by BrettDong** [comment]
```
ccache g++  -DRELEASE -DGIT_VERSION -DLOCALIZE -D__extern_always_inline="extern __always_inline" -ffast-math -Os  -Wall -Wextra    -fsigned-char -Werror -std=c++11 -MMD -MP -I/usr/include/ncursesw -c src/game_inventory.cpp -o obj/game_inventory.o
make: *** [obj/game_inventory.o] Killed
make: *** Waiting for unfinished jobs....
```

Travis CI out of memory? Close & reopen to trigger a rebuild.

**Comment by BrettDong** [comment]
@BevapDin Are _requested changes_ ready now?

File: src/output.cpp [hunk 4]
```diff
@@ -1756,28 +1756,21 @@ get_hp_bar( const int cur_hp, const int max_hp, const bool is_mon )
...
+         {
+             pair_t {translate_marker( "unknown" ), c_pink},
+             pair_t {translate_marker( "bright" ), c_yellow},
+             pair_t {translate_marker( "cloudy" ), c_white},
+             pair_t {translate_marker( "shady" ), c_ltgray},
+             pair_t {translate_marker( "dark" ), c_dkgray},
+             pair_t {translate_marker( "very dark" ), c_black_white}
+         }
+     };
+    const int light_level = clamp( ( int )ceil( light ), 0, 5 );
```
Comment (codemime) [inline src/output.cpp]: Please try not to use "magic numbers".  `strings.size() - 1` is more robust and clearly states the meaning. Optionally `static_assert(!strings.empty())` can be placed before accessing the array by the clamped index; Also, `( int )` is a C-ish way of casting which is not recommended: hurts type-safety, but kept for compatibility purposes. `static_cast`'s should be used instead. Also, array indexes are essentially `size_t`s, not `int`s.

**Pull request merged**